### DISCUSSION TITLE
Query on Message Queue support

### DISCUSSION THREAD

**Discussion by supriyatech-hub** [apache/skywalking#9678] [Q&A] [answered]
Hi,

I was going through the documentation and found this link which mentions about using message queue.
https://skywalking.apache.org/docs/main/next/en/setup/backend/mq/#message-queue-performance-and-consuming-latency-monitoring

Does it mean that Skywalking supports use of message queues for communication between skywalking components?

**Comment by Superskyyy** [comment, marked as answer]
@supriyatech-hub 
Hello, I believe that you understood it wrongly. The docs page is about how SkyWalking monitors MQ systems instead of using MQ to communicate between components (if you meant by such as an agent <-> SkyWalking backend).

But we do have support for MQs for communication.

Read these two following docs to understand MQs in SkyWalking, and how to use Kafka fetcher/ other MQ fetchers (rockerMQ and Pulsar in SkyAPM integration)
https://skywalking.apache.org/docs/main/next/en/faq/why_mq_not_involved/ 
https://skywalking.apache.org/docs/skywalking-satellite/next/en/concepts-and-designs/overview/
https://skywalking.apache.org/docs/main/next/en/setup/backend/kafka-fetcher/#kafka-fetcher
